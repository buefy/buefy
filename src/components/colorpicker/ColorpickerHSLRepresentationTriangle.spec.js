import { shallowMount } from '@vue/test-utils'
import Color from '@utils/color'
import BColorpickerHSLRepresentationTriangle from '@components/colorpicker/ColorpickerHSLRepresentationTriangle'

let defaultProps

describe('BColorpickerHSLRepresentationTriangle', () => {
    beforeEach(() => {
        defaultProps = () => ({
            value: Color.parse('#123456')
        })
    })
    it('render correctly', () => {
        const wrapper = shallowMount(BColorpickerHSLRepresentationTriangle, {
            propsData: {
                ...defaultProps()
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
})
