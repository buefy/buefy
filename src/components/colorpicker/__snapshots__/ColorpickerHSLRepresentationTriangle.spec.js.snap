// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BColorpickerHSLRepresentationTriangle render correctly 1`] = `
<svg viewBox="0 0 200 200" class="b-colorpicker-triangle">
    <defs>
        <linearGradient id="cp-triangle-gradient-ligthness-0" x1="0" y1="0" x2="1" y2="0">
            <stop offset="0%" stop-color="#fff"></stop>
            <stop offset="100%" stop-color="#000"></stop>
        </linearGradient>
        <linearGradient id="cp-triangle-gradient-saturation-0" x1="0" y1="0" x2="0" y2="1">
            <stop offset="0%" stop-color="hsl(210deg, 100%, 50%)" stop-opacity="1"></stop>
            <stop offset="100%" stop-color="hsl(210deg, 100%, 50%)" stop-opacity="0"></stop>
        </linearGradient>
        <clipPath id="cp-triangle-clip-0">
            <path d="M2 100a98  98  0 1 1 196 0h-20a-78  78  0 1 0 -156 0a78  78  0 1 0 156 0h20a98  98  0 1 1 -196 0z"></path>
        </clipPath>
    </defs>
    <g class="colorpicker-triangle-slider-hue">
        <foreignObject x="0" y="0" width="200" height="200" clip-path="url(#cp-triangle-clip-0)">
            <div class="colorpicker-triangle-hue"></div>
        </foreignObject>
        <g style="transform: rotate(210deg);">
            <foreignObject x="96" y="0" width="8" height="24">
                <div role="slider" tabindex="0" aria-label="Hue" aria-valuemin="0" aria-valuenow="210" aria-valuemax="360" class="hue-range-thumb"></div>
            </foreignObject>
        </g>
    </g>
    <g role="graphics-datagroup" aria-datascales="lightness, saturation" class="colorpicker-triangle-slider-sl" style="transform: rotate(210deg) translate(50%, 50%);">
        <path d="M0 -78L67.54998149484 39H-67.54998149484z" fill="url(#cp-triangle-gradient-ligthness-0)"></path>
        <path d="M0 -78L67.54998149484 39H-67.54998149484z" fill="url(#cp-triangle-gradient-saturation-0)" style="mix-blend-mode: overlay;"></path>
        <foreignObject x="34.010373654636" y="-44.885000000000005" width="12" height="12">
            <div tabindex="0" aria-datavalues="67%, 20%" class="sl-range-thumb"></div>
        </foreignObject>
    </g>
</svg>
`;
